packages/hooks/src/createDoingHook.ts

```diff
@@ -1,24 +1,25 @@
 /**
- * @callback DoingHook
+ * Internal dependencies
+ */
+import type { Hooks, StoreKey } from '.';
+
+/**
  * Returns whether a hook is currently being executed.
  *
- * @param {string} [hookName] The name of the hook to check for.  If
- *                            omitted, will check for any hook being executed.
- *
- * @return {boolean} Whether the hook is being executed.
  */
+export type DoingHook = ( hookName?: string ) => boolean;
```

Comment:
Let us retain the docs here as well

```suggestion
export type DoingHook = (
	/**
	 * The name of the hook to check for.
	 * If omitted, will check for any hook being executed.
	 */
	hookName?: string
) => boolean;
```